app/views/shared/views/_retire.html.haml

```diff
@@ -1,83 +1,18 @@
-%form#form_div.form-horizontal{"name"          => "angularForm",
-                               "ng-controller" => "retirementFormController as vm",
-                               "miq-form"      => true,
-                               "model"         => "vm.retirementInfo",
-                               "model-copy"    => "vm.modelCopy",
-                               "form-changed"  => true }
-  %h3
-    = _('Set/Remove Retirement Date')
-  %tbody
-    .form-group
-      %label.col-md-3.control-label
-        = _('Enter Retirement Date as')
-      .col-md-4{"ng-init" => "formMode = 'date'"}
-        = select_tag("formMode",
-                     options_for_select(_("Specific Date and Time") => 'date', _("Time Delay from Now") => 'delay'),
-                     "ng-model"  => "formMode",
-                     "pf-select" => true)
-    .form-group{"ng-if" => "formMode == 'date'"}
-      %label.col-md-3.control-label
-        = _('Retirement Date and Time')
-      .col-md-4#retirement_date_div
-        = datetimepicker_input_tag('retirement_date_datepicker', nil,
-                                   'class'      => 'form-control',
-                                   'ng-model'   => 'vm.retirementInfo.retirementDate',
-                                   'start-date' => 'vm.datepickerStartDate')
-    %datetime-delay-picker{'ng-if'      => "formMode == 'delay'",
-                           'model'      => 'vm.retirementInfo.retirementDate',
-                           'start-date' => 'vm.datepickerStartDate'}
-    .form-group{"ng-if" => "formMode == 'delay'"}
-      %label.col-md-3.control-label
-        = _('Retirement Date and Time')
-      .col-md-4#retirement_date_result_div
-        %input.form-control#retirement_date{'readonly' => true, 'ng-model' => 'vm.retirementInfo.retirementDate'}
-    .form-group
-      %label.col-md-3.control-label
-        = _('Retirement Warning')
-      .col-md-8
-        - select_options = [{:value => "", :label => _("None")},
-                            {:value => 7,  :label => _("1 Week before retirement")},
-                            {:value => 14, :label => _("2 Weeks before retirement")},
-                            {:value => 30, :label => _("30 Days before retirement")}]
-        = select_tag("retirementWarning",
-                     "",
-                     "ng-options"  => "item.value as item.label for item in #{select_options.to_json.gsub('"', '\'')}",
-                     "ng-model"    => "vm.retirementInfo.retirementWarning",
-                     "ng-disabled" => "!vm.retirementInfo.retirementDate",
-                     "pf-select"   => true)
-        :javascript
-          miqInitSelectPicker();
-  %div{"ng-if" => "formMode == 'date'"}
-    = _('* Saving a blank date will remove all retirement dates')
+%retirement-form{'object-ids' => @retireitems.collect(&:id)}
```

Comment:
`@retireitems.collect(&:id).map(&:to_s).to_json`

(So that we end up with `object-ids="[ &quot;123&quot;, &quot;456&quot; ]"` instead of `object-ids="[123, 456]"` which are parsed as numbers.)